interceptors/logging/logging.go

```diff
@@ -38,6 +38,30 @@ func newCommonFields(kind string, c interceptors.CallMeta) Fields {
 	}
 }
 
+// disableCommonLoggingFields returns copy of common fields with disabled fields removed.
+func disableCommonLoggingFields(kind string, c interceptors.CallMeta, disableFields []string) Fields {
+	commonFields := newCommonFields(kind, c)
+
+	existing := map[any]any{}
```

Comment:
Can't this be of the `Fields` type? Then we don't have to do the final loop at the end.

Reply:
I took the reference from here -> https://github.com/grpc-ecosystem/go-grpc-middleware/blob/4679fb12b6915f8f7a682a525073fe3810d5c64e/interceptors/logging/logging.go#L81

Fields is of `type Fields []any` and delete does not work on that type

Reply:
I see. I think there are still a few problems with this solution:
1. The current solution will reorder elements in the fields (because of the map iteration), which could be confusing and annoying to users.
2. It might be easier to just populate `newFields` directly while iterating over `disableFields`.

I think the better solution here is actually to implement a new function on `Fields`: `Delete(k string)`. `Fields` is essentially pretending to be a map, so deleting by a key makes sense. The implementation of the `Delete` method would mutate the receiver directly by removing the element and the value immediately following it. Then this function can just loop over the disabledFields and call Delete.

What do you think?